endpoints/openrtb2/auction.go

```diff
@@ -668,11 +668,31 @@ func (deps *endpointDeps) validateImpExt(imp *openrtb.Imp, aliases map[string]st
 	if len(imp.Ext) == 0 {
 		return []error{fmt.Errorf("request.imp[%d].ext is required", impIndex)}
 	}
+
 	var bidderExts map[string]json.RawMessage
 	if err := json.Unmarshal(imp.Ext, &bidderExts); err != nil {
 		return []error{err}
 	}
 
+	// Also accept bidder exts within imp[...].ext.prebid.bidder
+	// NOTE: This is not part of the official API, we are not expecting clients
+	// migrate from imp[...].ext.${BIDDER} to imp[...].ext.prebid.bidder.${BIDDER}
+	// at this time
+	// https://github.com/prebid/prebid-server/pull/846#issuecomment-476352224
+	if rawPrebidExt, ok := bidderExts["prebid"]; ok {
+		var prebidExt openrtb_ext.ExtImpPrebid
+		if err := json.Unmarshal(rawPrebidExt, &prebidExt); err == nil && prebidExt.Bidder != nil {
+			for bidder, ext := range prebidExt.Bidder {
+				if ext == nil {
+					continue
+				}
+
+				bidderExts[bidder] = ext
```

Comment:
Since with this PR, bidder extensions can possibly be in two places (`imp[...].ext.prebid.bidder` or `imp[...].ext.bidder`), are we assuming that `imp[...].ext.prebid.bidder` will always override `imp[...].ext.bidder`? I do understand that ideally this shouldn't happen but just wondering what if it does.

Reply:
yup exactly, but as you said this should never happen. Also, in reality only impbus is going to be using ext.prebid.bidder for the foreseeable future